.github/setup-env/action.yml

```diff
@@ -0,0 +1,28 @@
+name: Setup Python environment
+description: Set up Python environment for the repository
+inputs:
+  python-version:
+    description: "The Python version to set up"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v4
+      with:
+        # Install a specific version of uv.
+        version: "0.5.9"
+
+    - name: Install hssm
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: uv sync --all-extras
```

Comment:
This is my fault. We only need to run `uv sync`. We have an optional dependency group that installs JAX with GPU support, which is unsupported in the dev environment. Let's not install those dependencies.